code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,142 @@
+//I JUST WANNA GRILL FOR GOD'S SAKE
+
+/obj/machinery/grill
+	name = "grill"
+	desc = "Just like the old days."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grill_open"
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/obj/item/reagent_containers/food/grilled_item
+	var/grill_time = 0
+	var/datum/looping_sound/grill/grill_loop
+
+/obj/machinery/grill/Initialize()
+	. = ..()
+	grill_loop = new(list(src), FALSE)
+
+/obj/machinery/grill/update_icon()
+	if(grilled_item)
+		icon_state = "grill"
+	else if(grill_fuel)
+		icon_state = "grill_on"
+	else
+		icon_state = "grill_open"
+
+/obj/machinery/grill/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack/sheet/mineral/coal) || istype(I, /obj/item/stack/sheet/mineral/wood))
+		var/obj/item/stack/S = I
+		var/stackamount = S.get_amount()
+		to_chat(user, "<span class='notice'>You put [stackamount] [I]s in [src].</span>")
+		if(istype(I, /obj/item/stack/sheet/mineral/coal))
+			grill_fuel += (500 * stackamount)
+		else
+			grill_fuel += (50 * stackamount)
+		S.use(stackamount)
+		update_icon()
+		return
+	if(I.resistance_flags & INDESTRUCTIBLE)
+		to_chat(user, "<span class='warning'>You don't feel it would be wise to grill [I]...</span>")
+		return ..()
+	if(istype(I, /obj/item/reagent_containers))
```

Comment:
what does this check do? Why couldn't you just do the below check here?

Reply:
you can add the monkey energy reagent as fuel, and you need a reagent container to do that
just look at the else LOL!!